### PR TITLE
feat(perf-detector-threshold-configuration) Re-added detection enabled/disabled code

### PR DESCRIPTION
**Pull request by Abdkhan14**
Last step in the data migration process.
New functionality was already approved here: https://github.com/getsentry/sentry/pull/50825

- Manually set enabled/disabled values based on existing detection rates.
- Removed detection_rate functionality.
- Used detect_enabled functionality.
- Added back, commented out tests. 
- Removed an overlapping test from test_performance_detection.py

### CODE REVIEW

File: tests/sentry/utils/performance_issues/test_performance_detection.py [hunk 1]
```diff
@@ -129,22 +129,6 @@ def test_project_option_overrides_default(self):
...
-
-    #     perf_problems = _detect_performance_problems(n_plus_one_event, sdk_span_mock, self.project)
-    #     assert_n_plus_one_db_problem(perf_problems)
-
-    #     self.project_option_mock.return_value = {
-    #         "n_plus_one_db_queries_detection_enabled": False,
-    #     }
-
-    #     perf_problems = _detect_performance_problems(n_plus_one_event, sdk_span_mock, self.project)
-    #     assert perf_problems == []
```
Comment (narsaynorath) [inline tests/sentry/utils/performance_issues/test_performance_detection.py:146]: Did you mean to remove this test?

**Review by narsaynorath — APPROVED** [review]
lgtm, but I noticed a test was removed when the comment said to just uncomment it

File: tests/sentry/utils/performance_issues/test_performance_detection.py (hunk 1 above)
Comment (Abdkhan14) [reply tests/sentry/utils/performance_issues/test_performance_detection.py:146]: Yeah, I mentioned it the PR description. It's an overlapping test. Will add a better test in the coming PRs.

Comment (narsaynorath) [reply tests/sentry/utils/performance_issues/test_performance_detection.py:146]: Oops, I missed that! Thanks for clarifying

**Pull request merged**